driver-sync/src/main/com/mongodb/client/internal/ClientEncryptionImpl.java

```diff
@@ -171,6 +183,65 @@ public RewrapManyDataKeyResult rewrapManyDataKey(final Bson filter, final Rewrap
         return new RewrapManyDataKeyResult(collection.bulkWrite(updateModels));
     }
 
+    @Override
+    public BsonDocument createEncryptedCollection(final MongoDatabase database, final String collectionName,
+            final CreateCollectionOptions createCollectionOptions, final CreateEncryptedCollectionParams createEncryptedCollectionParams)
+            throws MongoUpdatedEncryptedFieldsException {
+        notNull("collectionName", collectionName);
+        notNull("createCollectionOptions", createCollectionOptions);
+        notNull("createEncryptedCollectionParams", createEncryptedCollectionParams);
+        MongoNamespace namespace = new MongoNamespace(database.getName(), collectionName);
+        Bson rawEncryptedFields = createCollectionOptions.getEncryptedFields();
+        if (rawEncryptedFields == null) {
+            throw new MongoConfigurationException(format("`encryptedFields` is not configured for the collection %s.", namespace));
+        }
+        CodecRegistry codecRegistry = options.getKeyVaultMongoClientSettings() == null
+                ? MongoClientSettings.getDefaultCodecRegistry()
+                : options.getKeyVaultMongoClientSettings().getCodecRegistry();
+        BsonDocument actualEncryptedFields = rawEncryptedFields.toBsonDocument(BsonDocument.class, codecRegistry).clone();
```

Comment:
If a user used a `RawBsonDocument` that might trip us up here.

I'm not sure if we need to guard against that, its just its immutable and will error later.

Reply:
Thanks! Given that `RawBsonDocument` is immutable, the best solution here appears to fix the `RawBsonDocument.clone` method. Currently, despite the return type of the method being `BsonDocument`, the method returns `RawBsonDocument`. Moreover, it creates the new `RawBsonDocument` by cloning `RawBsonDocument.bytes`, which, as far as I can see, has no effect beyond wasting resources.

We can change the implementation of `RawBsonDocument.clone` such that it returns `toBaseBsonDocument()` - a normal mutable `BsonDocument`. Given that currently `RawBsonDocument.clone` is not documented beyond what `Object.equals` says, this is not a breaking change. What do you think?

Reply:
Lets do this work in a separate ticket.

Reply:
Submitted https://github.com/mongodb/mongo-java-driver/pull/1081 for review.